build/gulp/tasks/perf.ts

```diff
@@ -0,0 +1,117 @@
+import * as express from 'express'
+import * as fs from 'fs'
+import { task, series } from 'gulp'
+import * as _ from 'lodash'
+import * as puppeteer from 'puppeteer'
+import * as rimraf from 'rimraf'
+import { argv } from 'yargs'
+
+import { ProfilerMeasure, ProfilerMeasureCycle } from '../../../perf/types'
+import config from '../../../config'
+import webpackPlugin from '../plugins/gulp-webpack'
+
+const { paths } = config
+const { colors, log } = require('gulp-load-plugins')().util
+
+let server
+
+const floor = (value: number) => _.floor(value, 2)
+
+const computeMeasureMedian = (measures: ProfilerMeasure[], key: string) => {
+  const values = _.sortBy(measures, key)
+
+  const lowMiddle = Math.floor((values.length - 1) / 2)
+  const highMiddle = Math.ceil((values.length - 1) / 2)
+
+  return (values[lowMiddle][key] + values[highMiddle][key]) / 2
+}
+
+const reduceMeasures = (measures: ProfilerMeasure[], key: string) => {
+  let min = measures[0][key]
+  let max = measures[0][key]
+  let sum = measures[0][key]
+
+  for (let i = 1; i < measures.length; i++) {
+    if (measures[i][key] < min) min = measures[i][key]
+    if (measures[i][key] > max) max = measures[i][key]
+
+    sum = sum + measures[i][key]
+  }
+
+  return {
+    avg: floor(sum / measures.length),
+    median: floor(computeMeasureMedian(measures, key)),
+    min: floor(min),
+    max: floor(max),
+    values: _.map(measures, measure => ({
+      order: measure.order,
+      value: measure[key],
+    })),
+  }
+}
+
+const normalizeMeasures = (measures: ProfilerMeasureCycle[]) => {
+  const perExampleMeasures = _.reduce(
+    measures,
+    (result, cycle: ProfilerMeasureCycle) => {
+      _.forEach(cycle, (value: ProfilerMeasure, exampleName: string) => {
+        result[exampleName] = [...(result[exampleName] || []), value]
+      })
+
+      return result
+    },
+    {},
+  )
+
+  return _.mapValues(perExampleMeasures, (measures: ProfilerMeasure[]) => ({
+    actualTime: reduceMeasures(measures, 'actualTime'),
+    baseTime: reduceMeasures(measures, 'baseTime'),
+  }))
+}
+
+task('perf:clean', cb => {
+  rimraf(paths.perfDist(), cb)
+})
+
+task('perf:build', cb => {
+  webpackPlugin(require('../../../build/webpack.config.perf').default, cb)
+})
+
+task('perf:run', async () => {
+  const measures: ProfilerMeasureCycle[] = []
+  const times = argv.times || 10
+
+  const browser = await puppeteer.launch()
+
+  for (let i = 0; i < times; i++) {
+    const page = await browser.newPage()
+    await page.goto(`http://${config.server_host}:${config.server_port}`)
+
+    const measuresFromStep = await page.evaluate(() => window.runMeasures())
+    await page.close()
+
+    measures.push(measuresFromStep)
+  }
+
+  await browser.close()
+
+  fs.writeFileSync(
+    paths.perfDist('result.json'),
```

Comment:
lets create some console log entry about that or, much better, display results to the console somehow. Currently it is completely non obvious to the client where these results could be found :)

Reply:
Added log entry:
```
[16:31:18] Results are written to "...PATH...\react\perf\dist\result.json"
```